clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -299,6 +294,13 @@ protected synchronized void bootstrap(Optional<Set<String>> managedClients) {
         storeNameToPartitionListMap.put(storeName, storageService.getUserPartitions(kafkaTopicName));
       }
     }
+    /**
```

Comment:
Why did we need to move this? If a store/version is deleted, the previous snippet will fail fast and the execution will not reach here, right?
```
Version version = storeRepository.getStoreOrThrow(storeName)
          .getVersion(versionNumber)
          .orElseThrow(
              () -> new VeniceException(
                  "Could not find version: " + versionNumber + " for store: " + storeName + " in storeRepository!"));
```

Reply:
Hmm I think it is not related...so what I want to do here in the big loop is:
1. Scan all storage engines, remove unused storage engines and mark those unused stores.
2. For useful storage engines,  compute the useful(valid) versions to bootstrap and set the target version based on local state.

The part of logic I moved is to remove StoreBackend object for unused store which was registered during scan. I think we should only do it once at the end.

Reply:
Just went through the code again. I agree that the previous removal of unusedStores looks incorrectly placed. Resolving the comment